src/Core/System/Currency/Api/ChangeCurrencyController.php

```diff
@@ -0,0 +1,143 @@
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"api"})
+ */
+class ChangeCurrencyController
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var DefinitionInstanceRegistry
+     */
+    private $definitionInstanceRegistry;
+
+    /**
+     * @var MessageBusInterface
+     */
+    private $messageBus;
+
+    /**
+     * @var CacheClearer
+     */
+    private $cacheClearer;
+
+    public function __construct(
+        Connection $connection,
+        DefinitionInstanceRegistry $definitionInstanceRegistry,
+        MessageBusInterface $messageBus,
+        CacheClearer $cacheClearer
+    ) {
+        $this->connection = $connection;
+        $this->definitionInstanceRegistry = $definitionInstanceRegistry;
+        $this->messageBus = $messageBus;
+        $this->cacheClearer = $cacheClearer;
+    }
+
+    /**
+     * @Route(path="/api/v{version}/_action/currency/change-default-currency/{currencyId}", methods={"POST"}, name="api.custom.currency.change-default")
+     */
+    public function change(string $currencyId): JsonResponse
+    {
+        $currencyId = strtolower($currencyId);
+        $this->checkIsValidCurrencyId($currencyId);
+
+        $defaultCurrencyId = Uuid::fromHexToBytes(Defaults::CURRENCY);
+
+        $swapId = Uuid::randomBytes();
+
+        $stmt = $this->connection->prepare('UPDATE currency SET id = :newId WHERE id = :oldId');
+
+        $stmt->execute([
+            'newId' => $swapId,
+            'oldId' => $defaultCurrencyId,
+        ]);
+
+        $stmt->execute([
+            'newId' => $defaultCurrencyId,
+            'oldId' => Uuid::fromHexToBytes($currencyId),
+        ]);
+
+        $factor = (float) $this->connection->fetchColumn('SELECT factor FROM currency WHERE id = :id', ['id' => $defaultCurrencyId]);
+        $this->connection->executeQuery(
+            'UPDATE currency SET factor = IF(id = :id, 1, factor * :fixFactor);',
+            [
+                'id' => $defaultCurrencyId,
+                'fixFactor' => 1 / $factor,
+            ]
+        );
+
+        $this->schedulePriceUpdates($factor);
+
+        $this->cacheClearer->invalidateTags([
+            'entity_currency', // Invalidate entity reader
+            'currency.id', // Invalidate entity searcher
+        ]);
+
+        return new JsonResponse(['id' => Defaults::CURRENCY]);
+    }
+
+    private function checkIsValidCurrencyId(string $currencyId): void
+    {
+        if ($currencyId === Defaults::CURRENCY) {
+            throw new \RuntimeException('The given currencyId is already the default');
+        }
+
+        $exists = (bool) $this->connection->fetchColumn('SELECT 1 FROM currency WHERE id = UNHEX(?) LIMIT 1', [
+            $currencyId,
+        ]);
+
+        if (!$exists) {
+            throw new \RuntimeException('The given currencyId does not exists');
+        }
+    }
+
+    private function schedulePriceUpdates(float $fixFactor): void
```

Comment:
There is a special case missing. We need to update `shipping_method_price.price` manually if `shopware.deployment.blue_green === false` - that means no trigger. Background: We deprecated a float price + currency id and replaced it with a price field.